### DISCUSSION TITLE
avalonia11-preview5 compatibility

### DISCUSSION THREAD

**Discussion by graealex** [AvaloniaCommunity/Material.Avalonia#213] [General]
When is compatibility with avalonia11-preview5 going to happen? Preview 5 introduced a number of breaking changes, so whoever is on preview 4 right now (like me) has quite the incentive to switch to preview 5 as soon as possible. Right now, when using Material.Avalonia, it's not possible to switch to preview 5 due to incompatibilities in Material.Avalonia, even in the avalonia11preview5 branch when compiling myself.

**Comment by SKProCH** [comment]
Hello, yep, i'm working at this problem. 
Actually this is almost complete at this [branch](https://github.com/AvaloniaCommunity/Material.Avalonia/tree/avalonia11preview5), i just need to fix the styles initialization. I think what i can complete this at this week. It's the last week of my study session, so I have almost no time. If it's very important to you, you can always do PR. Or wait, I think it will be ready by the end of the week.

**Reply by graealex** [reply]
I did try my luck, but the remaining problems are actually beyond my insight into what actually changed. I understand that preview 5 had some major changes regarding theming. Well, I hope you can find some time. I would have sent you a tip, but currently it's literally impossible - neither VK nor PayPal, sorry

**Reply by SKProCH** [reply]
I'll try to fix this tomorrow, actually. Hope i'll find time

**Reply by SKProCH** [reply]
Fixed MaterialToolkit resources lookup, but there is some other problems. For example, ItemsPresenter does not have Items property, and i can't just replace it with ItemsControl. This is need investigation

**Reply by graealex** [reply]
The channel said to look at https://github.com/AvaloniaUI/Avalonia/blob/7dbf33e043edf79371c3e70659375ec202fe3682/src/Avalonia.Themes.Fluent/Controls/ItemsControl.xaml#L11-L14 

The explanation was:
It's looking for the items in the parent iirc
ItemsPresenter was always meant to be an implementation detail of ItemsControl and similar components
it shouldn't be used separately

Maybe that helps.

**Comment by SKProCH** [comment]
Nightly release `3.0.0-avalonia11-preview3.145-nightly` with preview-5 support published

**Reply by SKProCH** [reply]
Current preview-5 TODO list: https://github.com/AvaloniaCommunity/Material.Avalonia/issues/217

**Reply by graealex** [reply]
Seems to work. Thanks!
